src/llvm-codegen-shared.h

```diff
@@ -2,26 +2,21 @@
 
 #include <utility>
 #include <llvm/ADT/ArrayRef.h>
-#if JL_LLVM_VERSION >= 160000
-#include <llvm/ADT/Optional.h>
-#endif
 #include <llvm/Support/Debug.h>
 #include <llvm/IR/Attributes.h>
 #include <llvm/IR/DebugLoc.h>
 #include <llvm/IR/IRBuilder.h>
 #include <llvm/IR/MDBuilder.h>
+
+#if JL_LLVM_VERSION >= 160000
 #include <llvm/Support/ModRef.h>
+#endif
+
 #include "julia.h"
 
 #define STR(csym)           #csym
 #define XSTR(csym)          STR(csym)
 
-#if JL_LLVM_VERSION >= 160000
-namespace llvm {
```

Comment:
@pchintalapudi why did you delete this?

See #51720

Reply:
I got this working by keeping this define here, making jitlayers.h include codegen-shared and remove the places where we include both (we can also use `#pragma once`)